### DISCUSSION TITLE
Does WG_ALLOWED_IPS work?

### DISCUSSION THREAD

**Discussion by eduardogsilva** [wg-easy/wg-easy#823] [General]
Hello there! 

I'm trying to run wg-easy for connecting multiple sites.
I've tried to declare WG_ALLOWED_IPS to allow routing to internal networks.
I don't see the result of this parameter in any configuration file, even if I declare WG_ALLOWED_IPS with 0.0.0.0/0 (or if I list the private networks like the example below).

If I open the wg0.conf or wg0.json, I can only see the designated ip addess/32 for each customer in the allowedips field.

docker run -d \
  --name=wg-easy \
  -e LANG=en \
  -e WG_HOST=mydomain.com \
  -e PASSWORD=secret \
  -e WG_DEFAULT_ADDRESS=10.99.90.x \
  -e WG_PERSISTENT_KEEPALIVE=25 \
  -e WG_ALLOWED_IPS="192.168.0.0/16, 10.0.0.0/8, 172.16.0.0/12" \
  -v ~/.wg-easy:/etc/wireguard \
  -p 51820:51820/udp \
  -p 127.0.0.1:51821:51821/tcp \
  --cap-add=NET_ADMIN \
  --cap-add=SYS_MODULE \
  --cap-add=CAP_NET_BROADCAST \
  --sysctl="net.ipv4.conf.all.src_valid_mark=1" \
  --sysctl="net.ipv4.ip_forward=1" \
  --restart unless-stopped \
  ghcr.io/wg-easy/wg-easy

Maybe I've got something wrong?
Any ideas or sugestions?

Thank you

**Comment by eandersons** [comment]
After changing value of the environment variable `WG_ALLOWED_IPS`, the new value will be applied only to the newly created client configuration, the existing peers will still have their initial allowed IPs.

**Reply by eduardogsilva** [reply]
Hello Edgars,

Thank you very much for your input.

I might be missing something, but unfortunately, I'm not experiencing the expected outcome. Regardless of the configurations set in WG_ALLOWED_IPS (initially, I tried using 0.0.0.0/0, but to no avail), the issue persists.

Even after removing the container and starting again with the same command listed above and adding a new client, the AllowedIPs field in wg0.conf file continues to only show the /32 address indicated in the web interface.

Would you have any other suggestions?

Thank you once again for your assistance. :)

**Reply by eandersons** [reply]
`WG_ALLOWED_IPS` does not affect server side configuration (`wg0.conf`), only clients' configuration.

**Reply by eduardogsilva** [reply]
As far as I can understand, the server side configuration (wg0.conf), is where I need to add the extra ips allowed for the client, so I can route subnets between clients. 

Am I getting this wrong? because at this moment, I have connection between clients, and even if add static routes between the clients, it does not get routed properly to the subnet behind client.

Thank you once again :)

**Reply by eandersons** [reply]
Sorry, for your specific use case I will be of no use.

You can change `wg0.conf` manually, but this file will be overwritten after the next container restart.
You could try to search in issues or discussions if someone has tried to do something similar

**Reply by eduardogsilva** [reply]
Hello Edgars! Thanks a lot for all the help. 

I've made a bunch of tests and checked the wireguard website. I just noticed that what I'm looking for is exactly "Cryptokey routing", it's on their main page. -> https://www.wireguard.com/#cryptokey-routing

Adding a subnet to AllowedIPs on wg0.conf will allow packets being routed to and from the networks behind that specific client.

I've manually edited my wg0.conf and restarted wireguard. The routing between clients started to work immediately.

I guess that at at this moment, this is not possible using wg-easy. 

Edgars, thank you very much for your support. :)

**Comment by K4rlosReyes** [comment]
Same issue here, I've tried several ways to add `WG_ALLOWED_IPS` and none of them have worked... Trying to set up a site-to-site VPN Server.

**Reply by eduardogsilva** [reply]
Not sure if it's ok to write this here, but since last week I decided to write my own webui for wireguard with site to site scenarios in mind.
Just take a look on my profile, it's the wireguard_webadmin repo. 😉

**Reply by K4rlosReyes** [reply]
I was taking a look at the code and I noticed that they don't refer to the `WG_ALLOWED_IPS` when saving the config file:

```
  async __saveConfig(config) {
    let result = `
# Note: Do not edit this file directly.
# Your changes will be overwritten!

# Server
[Interface]
PrivateKey = ${config.server.privateKey}
Address = ${config.server.address}/24
ListenPort = 51820
PreUp = ${WG_PRE_UP}
PostUp = ${WG_POST_UP}
PreDown = ${WG_PRE_DOWN}
PostDown = ${WG_POST_DOWN}
`;

    for (const [clientId, client] of Object.entries(config.clients)) {
      if (!client.enabled) continue;

      result += `

# Client: ${client.name} (${clientId})
[Peer]
PublicKey = ${client.publicKey}
PresharedKey = ${client.preSharedKey}
AllowedIPs = ${client.address}/32`;
    }
```

Now I added this:

```
AllowedIPs = ${client.address}/32,${WG_ALLOWED_IPS}`;
```

And it worked.

**Reply by saiinc** [reply]
I'm tryed this trick: https://github.com/wg-easy/wg-easy/issues/527#issuecomment-1670613029
Just edit .wg-easy/wg0.json
"address": "192.168.1.0/24,10.8.0.2",
Add allowed network with netmask before peer ip and restart container.
